### PR TITLE
core: copy the SchemaDescriptor when rebuilding descriptor

### PR DESCRIPTION
**Pull request by herbyderby**
useMarshalledMessages works by duplicating a ServerServiceDefinition while replacing just the marshallers. It currently does not copy over the SchemaDescriptors, which breaks at least the ProtoReflectionService.

### CODE REVIEW

**Comment by zhangkun83** [comment]
LGTM

File: api/src/main/java/io/grpc/ServerInterceptors.java [hunk 1]
```diff
@@ -187,13 +187,15 @@ public InputStream parse(final InputStream stream) {
       wrappedMethods.add(wrapMethod(definition, wrappedMethodDescriptor));
```
Comment (zhangkun83) [inline api/src/main/java/io/grpc/ServerInterceptors.java]: `wrappedMethods` is never used, and this is causing the CI error.

Comment (ejona86) [reply api/src/main/java/io/grpc/ServerInterceptors.java]: Look like the earlier for loop should not have been deleted and should occur after `serviceBuilder` is created (just like before).

Comment (herbyderby) [reply api/src/main/java/io/grpc/ServerInterceptors.java]: Sorry, not sure what happened--some patching problem on my end. Should look something like this:

```
    // Build the new service descriptor
    final ServiceDescriptor.Builder serviceDescriptorBuilder =
        ServiceDescriptor.newBuilder(serviceDef.getServiceDescriptor().getName())
            .setSchemaDescriptor(serviceDef.getServiceDescriptor().getSchemaDescriptor());
    // Create the new service definition.
    final ServerServiceDefinition.Builder serviceBuilder =
        ServerServiceDefinition.builder(serviceDescriptorBuilder.build());
    for (ServerMethodDefinition<?, ?> definition : wrappedMethods) {
      serviceBuilder.addMethod(definition);
    }
    return serviceBuilder.build();
```

Comment (zhangkun83) [reply api/src/main/java/io/grpc/ServerInterceptors.java]: Applied fix.

**Pull request merged**